<?xml version="1.0"?>
<robot name="ur5">
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5/base_link.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5/base_link.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5/shoulder_link.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5/shoulder_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="shoulder_pan_joint" type="revolute">
    <origin xyz="0 0 0.0099916" rpy="1.5708 1.5708 0"/>
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <axis xyz="1 0 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="300" velocity="3"/>
  </joint>
  <link name="upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5/upper_arm_link.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5/upper_arm_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="shoulder_lift_joint" type="revolute">
    <origin xyz="-0.080008 0.053821 0" rpy="0 0 0"/>
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <axis xyz="0 -1 0"/>
    <limit lower="-1.57" upper="1.57" effort="200" velocity="3"/>
  </joint>
  <link name="forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5/forearm_link.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5/forearm_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="elbow_joint" type="revolute">
    <origin xyz="-0.425 0 0" rpy="0 0 0"/>
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <axis xyz="0 1 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="200" velocity="3"/>
  </joint>
  <link name="wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5/wrist_1_link.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5/wrist_1_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="wrist_1_joint" type="revolute">
    <origin xyz="-0.39225 0 0" rpy="0 0 0"/>
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <axis xyz="0 -1 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="200" velocity="3"/>
  </joint>
  <link name="wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5/wrist_2_link.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5/wrist_2_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="wrist_2_joint" type="revolute">
    <origin xyz="0 0.056149 0" rpy="0 0 0"/>
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <axis xyz="1 0 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="200" velocity="3"/>
  </joint>
  <link name="wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5/wrist_3_link.stl"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5/wrist_3_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="wrist_3_joint" type="revolute">
    <origin xyz="-0.09492 0.041547 0" rpy="3.1416 0 3.1416"/>
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <axis xyz="0 -1 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="100" velocity="3"/>
  </joint>
  <link name="vacuum_gripper1">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.0 0.0 0.0"/>
      </geometry>
    </collision>
    <gravity>0</gravity>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.02" length="0.015"/>
      </geometry>
    </visual>
  </link>
  <joint name="gripper_joint1" type="revolute">
    <axis xyz="1 0 0"/>
    <parent link="wrist_3_link"/>
    <child link="vacuum_gripper1"/>
    <origin rpy="1.570 0 0" xyz="0 0.04 0"/>
    <limit effort="50" velocity="50" lower="0" upper="0"/>
  </joint>
  <link name="camera_link">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
    </visual>
  </link>
  <joint name="camera_joint" type="fixed">
    <axis xyz="0 1 0"/>
    <origin xyz="0.03 0.04 0" rpy="0 0 1.508"/>
    <parent link="wrist_3_link"/>
    <child link="camera_link"/>
  </joint>
</robot>
